src/Spryker/Zed/AppPayment/Business/AppPaymentFacadeInterface.php

```diff
@@ -78,13 +80,36 @@ public function getPaymentPage(PaymentPageRequestTransfer $paymentPageRequestTra
      */
     public function handleWebhook(WebhookRequestTransfer $webhookRequestTransfer, WebhookResponseTransfer $webhookResponseTransfer): WebhookResponseTransfer;
 
+    /**
+     * Specification:
+     * - Requests the `AppPaymentPaymentMethodsPlatformPluginInterface::configurePaymentMethods()` method to return a list PaymentMethods to be added.
+     * - When the passed `AppPaymentPlatformPluginInterface` is not an instance of `AppPaymentPaymentMethodsPlatformPluginInterface` it will return early.
+     * - PaymentMethods that were already added will not be added again.
+     * - PaymentMethods that were already persisted and are no longer returned from `AppPaymentPaymentMethodsPlatformPluginInterface::configurePaymentMethods()` method will be deleted and trigger a `DeletePaymentMethod` message.
+     * - PaymentMethods that were already persisted and require an update will trigger a `UpdatePaymentMethod` message.
+     *
+     * @api
+     */
+    public function configurePaymentMethods(AppConfigTransfer $appConfigTransfer): AppConfigTransfer;
+
+    /**
+     * Specification:
+     * - Requests the `AppPaymentPaymentMethodsPlatformPluginInterface::configurePaymentMethods()` method to return a list PaymentMethods to add and to delete.
+     * - When the passed `AppPaymentPlatformPluginInterface` is not an instance of `AppPaymentPaymentMethodsPlatformPluginInterface` it will return early.
+     *
+     * @api
+     */
+    public function deletePaymentMethods(AppConfigTransfer $appConfigTransfer): AppConfigTransfer;
```

Comment:
These new Facade methods with input `AppConfigTransfer` are **confusing** (probably not only new, but also old), they are valid only as part of AppConfiguration plugins (`ConfigurationAfterSavePluginInterface, `ConfigurationAfterDelete...`,) and they cannot be used inside apps independently.

So what about removing them and using inside the plugins `$this->getFactory()->class->neededForPluginMethod()` ? All specification can be moved to Plugin description.

For an app project there is no difference on how to overwrite the logic, it's always via overriding Factory method.

So don't see a need in these Plugin-specific methods in the Facade.

Reply:
How should that work? Plugins are in the Communication layer and the logic is in the Business layer. The Plugin doesn't have access to the BusinessFactory. The CommunicationFactory should only create forms etc. but not business models.

We always have it like it is.

Reply:
Ouuugh... 
This is what I always didn't like in the Spryker architecture.

Not a request for changes, just my thoughts: 
We need to have some "CommunicationFacade" to do not pollute business Facade... or why Plugins are on communication layer if they work on business layer?...